pkgs/development/tools/continuous-integration/jenkins/default.nix

```diff
@@ -10,9 +10,19 @@ stdenv.mkDerivation rec {
     sha256 = "0lficvngxzl7q088n3ssnnhjicd0xxr0k3n0inz7pvjj27dl35rr";
   };
 
+  nativeBuildInputs = [ makeWrapper ];
+
   buildCommand = ''
-    mkdir -p "$out/webapps"
+    mkdir -p "$out/bin" "$out/webapps"
+
     cp "$src" "$out/webapps/jenkins.war"
+
+    # Create the `jenkins-cli` command.
+    ${openjdk}/bin/jar -xf "$src" WEB-INF/lib/cli-${version}.jar \
+      && mv WEB-INF/lib/cli-${version}.jar "$out/bin/jenkins-cli.jar"
```

Comment:
Maybe put the jenkins-cli.jar file in `$out/share/java`? From the [nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/):
```
JAR files that are intended to be used by other packages should be installed in $out/share/java. JDKs have a stdenv setup hook that add any JARs in the share/java directories of the build inputs to the CLASSPATH environment variable.
```

I'm not sure who else will be needing this .jar, but it sure feels odd to have it in `$out/bin`.

Reply:
The manual also suggests the directory `$out/share/package-name` for private JARs.